### PR TITLE
First draft works.

### PR DESCRIPTION
**Pull request by Ipuch**
Hi @TheoSabattie,
Here is the first version that works, could please make a review of the code I would truly appreciate.

### CODE REVIEW

**Comment by TheoSabattie** [comment]
You should use [BigUint](https://docs.rs/num-bigint/latest/num_bigint/) instead of i32 to prevent amount limitation

I'll read later other part ;)

**Comment by Ipuch** [comment]
> You should use [BigUint](https://docs.rs/num-bigint/latest/num_bigint/) instead of i32 to prevent amount limitation
> 
> I'll read later other part ;)

Just to be sure, what is the interest here of using BigUint? 
I understand it's meant to use big unisgned integers (?).

Because we expect ranks to be approximately between 1 and 200 (IMMO). Also, grades doesn't go beyond 8 from Excellent to Reject.

**Comment by TheoSabattie** [comment]
> I understand it's meant to use big unsigned integers (?).

Yes

> Just to be sure, what is the interest here of using BigUint?

i32 range is between: -2,147,483,648 to 2,147,483,647
u32 range is between: 0 to 4,294,967,295

=> Could be better to use u32 instead of i32

Why using BigUInt  ?
=> BigUInt has higher range.  
Your array contains "vote amount per mention" (NOT rank)  
=> Using bigUing allow to reduce vote amount limitation  

I assume you based your implementation on the [python library](https://github.com/MieuxVoter/majority-judgment-library-python/blob/main/majority_judgment/__init__.py)(?)

I based this quick review on the [java lib](https://github.com/MieuxVoter/majority-judgment-library-java)  
"But" maybe we don't need big range.  

I'll review totally your work at the end of the week ;)

File: src/main.rs [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+///
+/// # Example (panic)
+/// ```
+/// let mut poll_data = BTreeMap::new();
+/// poll_data.insert("Pizza", vec![0, 2, 3]);
+/// poll_data.insert("Chips", vec![0, 3, 2, 3, 4]);
+/// check_poll_length(&poll_data);
+/// ```
+///
+fn check_poll_length(poll_data: &BTreeMap<String, Vec<i32>>) -> Result<(), &str> {
```
Comment (TheoSabattie) [inline src/main.rs]: => La valeur de retour a l'air inutile (?)

```rust
fn check_poll_length(poll_data: &BTreeMap<String, Vec<i32>>)  {
// ...
```

**Comment by TheoSabattie** [comment]
Ok', I was a bit lost. I understand your doubt. I was thinking we were implementing the libs with the same way.  

On Java :   
Proposal, Array => index = mention, value = vote amount (=one insert **per mention**).   

Your way is:   
Proposal, Array => value = mention value (=one insert **per vote**).

Globally seems ok', I would just change a lot of i32 to usize OR u32 / u16 / u8 (no need for negative value)

We need to setup some unit tests on all library. Seem ok like that for the moment ;) 👍

**Review by TheoSabattie — APPROVED** [review]

File: src/main.rs (hunk 1 above)
Comment (Ipuch) [reply src/main.rs]: Oui, je ne savais pas comment faire autrement...

**Comment by Ipuch** [comment]
> Ok', I was a bit lost. I understand your doubt. I was thinking we were implementing the libs with the same way.
> 
> On Java : Proposal, Array => index = mention, value = vote amount (=one insert **per mention**).
> 
> Your way is: Proposal, Array => value = mention value (=one insert **per vote**).
> 
> Globally seems ok', I would just change a lot of i32 to usize OR u32 / u16 / u8 (no need for negative value)
> 
> We need to setup some unit tests on all library. Seem ok like that for the moment ;) 👍

Thanks Theo ! I can open a new PR in case you see anything is needed :)

**Pull request merged**